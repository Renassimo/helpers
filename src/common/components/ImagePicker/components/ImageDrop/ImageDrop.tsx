import Typography from '@mui/material/Typography';

import { DropzoneState } from 'react-dropzone';

const ImageDrop = ({ dropzone }: { dropzone: DropzoneState }) => {
  const { getRootProps, getInputProps, isDragAccept } = dropzone;
  const backgroundColor = isDragAccept ? 'blue' : '#fafafa';
  return (
    <div
      {...getRootProps({
        style: {
          padding: '32px',
          backgroundColor,
          border: '1px dashed gray',
          color: 'gray',
          borderRadius: '4px',
        },
      })}
    >
      <input {...getInputProps()} />
      <Typography>Drag and drop map image, or click to select...</Typography>
    </div>
  );
};

export default ImageDrop;
