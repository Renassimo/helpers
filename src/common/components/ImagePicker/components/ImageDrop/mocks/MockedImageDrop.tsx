const MockedImageDrop = jest.fn(() => {
  return <div>MockedImageDrop</div>;
});

export default MockedImageDrop;
