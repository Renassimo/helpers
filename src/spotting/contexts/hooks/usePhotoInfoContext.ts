import { useContext } from 'react';

import PhotoInfoContext from '@/spotting/contexts/PhotoInfoContext';

const usePhotoInfoContext = () => useContext(PhotoInfoContext);

export default usePhotoInfoContext;
