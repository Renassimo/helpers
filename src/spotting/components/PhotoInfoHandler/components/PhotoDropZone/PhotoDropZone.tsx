import Box from '@mui/material/Box';

import { useDropzone, FileWithPath } from 'react-dropzone';

import ImageDrop from '@/common/components/ImagePicker/components/ImageDrop';

import { PhotoActionType } from '@/spotting/types';

import usePhotoInfoContext from '@/spotting/contexts/hooks/usePhotoInfoContext';

const PhotoDropZone = () => {
  const { dispatch } = usePhotoInfoContext();

  const dropzone = useDropzone({
    onDrop: (files: FileWithPath[]) =>
      dispatch({ type: PhotoActionType.FILES_DROP, payload: files }),
  });

  return (
    <Box>
      <ImageDrop dropzone={dropzone} />
    </Box>
  );
};

export default PhotoDropZone;
