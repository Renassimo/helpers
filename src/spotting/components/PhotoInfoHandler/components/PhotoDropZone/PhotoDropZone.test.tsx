import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import { useDropzone } from 'react-dropzone';

import ImageDrop from '@/common/components/ImagePicker/components/ImageDrop';

import usePhotoInfoContext from '@/spotting/contexts/hooks/usePhotoInfoContext';

import MockedImageDrop from '@/common/components/ImagePicker/components/ImageDrop/mocks/MockedImageDrop';

import PhotoDropZone from './PhotoDropZone';

jest.mock('react-dropzone');
jest.mock('@/common/components/ImagePicker/components/ImageDrop');
jest.mock('@/spotting/contexts/hooks/usePhotoInfoContext');

describe('PhotoDropZone', () => {
  beforeEach(() => {
    (useDropzone as unknown as jest.Mock).mockImplementation(
      jest.fn(() => 'mocked-dropzone')
    );
    (ImageDrop as unknown as jest.Mock).mockImplementation(MockedImageDrop);
    (usePhotoInfoContext as unknown as jest.Mock).mockImplementation(
      jest.fn(() => ({ dispatch: 'mocked-dispatch' }))
    );
  });

  test('renders successfully', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(<PhotoDropZone />);
    // Assert
    expect(baseElement).toMatchSnapshot();
    expect(useDropzone).toBeCalledWith({ onDrop: expect.any(Function) });
    expect(ImageDrop).toBeCalledWith({ dropzone: 'mocked-dropzone' }, {});
    expect(usePhotoInfoContext).toBeCalledWith();
  });
});
