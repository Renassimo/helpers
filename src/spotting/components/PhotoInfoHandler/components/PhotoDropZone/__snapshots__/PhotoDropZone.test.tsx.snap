// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoDropZone renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div>
        MockedImageDrop
      </div>
    </div>
  </div>
</body>
`;
